### PR TITLE
Ingress Update

### LINKED ISSUE #91: Support for path with both ingressClass and annotations
#  Support for path with both ingressClass and annotations

### IngressClass Support for path

When ingressClass is set to `gce`, `alb`, or `nxs`, the path should be set to `/*` with Dgraph, otherwise, `/` can be used for the path.

### Annotation Support for path 
Additionally, the annotations of `kubernetes.io/ingress.class` still needs to be supported if google-load-balancer-controller is not used on GKE. 

### References
* https://github.com/kubernetes/ingress-gce
* https://cloud.google.com/kubernetes-engine/docs/concepts/ingress#controller_summary

### PR DESCRIPTION
**Pull request by darkn3rd**
# Features

This is a refresh fix some issue with the ingress resource templates

* examples
  * removed `aws-alb-ingress-controller` as it is no longer supported
  * added example aws-load-balancer-controller
     * tested aws-load-balancer-controller on EKS + addon with dgraph
* null host support
   * host can be omitted (set to `null`) which will default it to `*`.  This is now supported 
* select on ingressClass
   * in older API, annotations were used to specify ingress controller class; in new API ingressClass is used
   * support setting value based on either annotation or ingressClass
   * note that annotation is still used with gce on GKE 

# Testing

Ran through tests using `helm template` and example values.

```bash
# alb 
helm template "dg" ./charts/dgraph/ --values ./charts/dgraph/example_values/ingress/ingress-alb.yaml
# google load balancer
helm template "dg" ./charts/dgraph/ --values ./charts/dgraph/example_values/ingress/ingress-gce.yaml
# ingress-nginx (openresty)
helm template "dg" ./charts/dgraph/ --values ./charts/dgraph/example_values/ingress/ingress-nginx-grpc.yaml
helm template "dg" ./charts/dgraph/ --values ./charts/dgraph/example_values/ingress/ingress-nginx.yaml
# nginx-plus
helm template "dg" ./charts/dgraph/ --values ./charts/dgraph/example_values/ingress/kubernetes-ingress.yaml
```

## Null Host Test

```bash
helm template "dg" ./charts/dgraph/ --values -  <<EOF
zero:
  persistence:
    storageClass: ebs-sc
alpha:
  persistence:
    storageClass: ebs-sc
  service:
    type: ClusterIP
global:
  ingress:
    enabled: true
    ingressClassName: alb
    annotations:
      alb.ingress.kubernetes.io/scheme: internet-facing
      alb.ingress.kubernetes.io/target-type: ip
    alpha_hostname: null
EOF
```

Fixes #91, #92, #93

- - -
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/charts/94)

### CODE REVIEW

File: charts/dgraph/templates/alpha/ingress.yaml [hunk 1]
```diff
@@ -1,3 +1,22 @@
+{{- /* Generate ingress path */}}
+{{- define "path" -}}
+  {{- $path := "/" -}}
+  {{- if .Values.global.ingress.ingressClassName -}}
+    {{- if eq .Values.global.ingress.ingressClassName "gce" "alb" "nsx" }}
```
Comment (nswamy) [inline charts/dgraph/templates/alpha/ingress.yaml:5]: can you add comment for this conditional

File: charts/dgraph/templates/ratel/ingress.yaml [hunk 2]
```diff
@@ -1,3 +1,22 @@
+{{- /* Generate ingress path */}}
+{{- define "path" -}}
+  {{- $path := "/" -}}
+  {{- if .Values.global.ingress.ingressClassName -}}
+    {{- if eq .Values.global.ingress.ingressClassName "gce" "alb" "nsx" }}
```
Comment (nswamy) [inline charts/dgraph/templates/ratel/ingress.yaml:5]: can you add a comment line why we have different paths?

**Review by nswamy — APPROVED** [review]
LGTM, except a few necessary changes/nits. Please make those changes and feel free to merge

**Review by MichelDiz — APPROVED** [review]

File: charts/dgraph/templates/alpha/ingress.yaml (hunk 1 above)
Comment (darkn3rd) [reply charts/dgraph/templates/alpha/ingress.yaml:5]: The `path` varies depending on the ingress controller.  Some ingress controllers require `/`, while others require `/*`.  

Ingress of `gce` (GLBC), `alb` (ALB), and `nsx` (VMWare) will require `/*`, while others will require `/`.

File: charts/dgraph/templates/ratel/ingress.yaml (hunk 2 above)
Comment (darkn3rd) [reply charts/dgraph/templates/ratel/ingress.yaml:5]: The `path` varies depending on the ingress controller.  Some ingress controllers require `/`, while others require `/*`.  

Ingress of `gce` (GLBC), `alb` (ALB), and `nsx` (VMWare) will require `/*`, while others will require `/`.

Comment (nswamy) [reply charts/dgraph/templates/ratel/ingress.yaml:5]: i meant a comment in the code

**Pull request merged**